lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -786,14 +786,17 @@ def remove_vm(self, vm):
             return {'changed': True, 'failed': False}
 
     def configure_guestid(self, vm_obj, vm_creation=False):
+        # guest_id is not required when using templates
+        if self.should_deploy_from_template():
```

Comment:
Please change this check to if self.params['guest_id'] is None and self.should_deploy_from_template()